ansible/library/reduce_and_add_sonic_images.py

```diff
@@ -169,13 +169,6 @@ def install_new_sonic_image(module, new_image_url, save_as=None):
             module.fail_json(msg="Image installation failed: rc=%d, out=%s, err=%s" %
                              (rc, out, err))
 
-    # If sonic device is configured with minigraph, remove config_db.json
```

Comment:
is this used only in the context of test_upgrade_path? otherwise- it might break other flows.

Reply:
@roy-sror , I have combined the change in this PR to https://github.com/sonic-net/sonic-mgmt/pull/10427 with another fix of the upgrade path test. And changed the logic to keep the config_db.json when doing upgrade instead of only remove it when doing downgrade. Please review.